dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/ElasticListener.java

```diff
@@ -228,16 +232,110 @@ private String getNDJSON(List<Entry> publishedEntries) {
      * @return The Elasticsearch object string to be placed into the index
      * @throws IOException  Mapper problems
      */
-    public static JsonNode dockstoreEntryToElasticSearchObject(Entry entry) throws IOException {
+    public static JsonNode dockstoreEntryToElasticSearchObject(final Entry entry) throws IOException {
         Set<Version> workflowVersions = entry.getWorkflowVersions();
         boolean verified = workflowVersions.stream().anyMatch(Version::isVerified);
         Set<String> verifiedPlatforms = getVerifiedPlatforms(workflowVersions);
-        JsonNode jsonNode = MAPPER.readTree(MAPPER.writeValueAsString(entry));
+        Entry detachedEntry = removeIrrelevantProperties(entry);
+        JsonNode jsonNode = MAPPER.readTree(MAPPER.writeValueAsString(detachedEntry));
         ((ObjectNode)jsonNode).put("verified", verified);
         ((ObjectNode)jsonNode).put("verified_platforms", MAPPER.valueToTree(verifiedPlatforms));
         return jsonNode;
     }
 
+    /**
+     * Remove some stuff that should not be indexed by ES.
+     * This is not ideal, we should be including things we want indexed, not removing.
+     * @param entry
+     */
+    private static Entry removeIrrelevantProperties(final Entry entry) {
+        Entry detachedEntry;
+        if (entry instanceof Tool) {
+            Tool tool = (Tool) entry;
+            Tool detachedTool = new Tool();
+            tool.getWorkflowVersions().forEach(version -> {
+                Hibernate.initialize(version.getSourceFiles());
+            });
+
+            // These are for facets
+            detachedTool.setDefaultWdlPath(tool.getDefaultWdlPath());
+            detachedTool.setDefaultCwlPath(tool.getDefaultCwlPath());
+            detachedTool.setNamespace(tool.getNamespace());
+            detachedTool.setRegistry(tool.getRegistry());
+            detachedTool.setPrivateAccess(tool.isPrivateAccess());
+
+            // These are for table
+            detachedTool.setGitUrl(tool.getGitUrl());
+            detachedTool.setName(tool.getName());
+            detachedTool.setToolname(tool.getToolname());
+            detachedEntry = detachedTool;
+        } else if (entry instanceof BioWorkflow) {
+            BioWorkflow bioWorkflow = (BioWorkflow) entry;
+            BioWorkflow detachedBioWorkflow = new BioWorkflow();
+            // These are for facets
+            detachedBioWorkflow.setDescriptorType(bioWorkflow.getDescriptorType());
+            detachedBioWorkflow.setSourceControl(bioWorkflow.getSourceControl());
+            detachedBioWorkflow.setOrganization(bioWorkflow.getOrganization());
+
+            // These are for table
+            detachedBioWorkflow.setWorkflowName(bioWorkflow.getWorkflowName());
+            detachedBioWorkflow.setRepository(bioWorkflow.getRepository());
+            detachedBioWorkflow.setGitUrl(bioWorkflow.getGitUrl());
+            detachedEntry = detachedBioWorkflow;
+        } else {
+            return entry;
```

Comment:
I guess there are few enough services at this point where this is not an issue? I feel like it should eventually be tackled, but I believe you're rewriting the whole thing for 1.9?

Reply:
Rewriting in the context of GitHub apps? I don't think that will affect this aspect of things

Reply:
I meant for ES itself -- I thought @garyluu mentioned something about changing the index or indexes. I don't know ES API at all, but it seems like you should be able to index the source files and still not have a huge response when requests are made from the client.